test/test_array_ops.cpp

```diff
@@ -316,6 +316,188 @@ TEST(ArrayOps, ExpandDims) {
 
 }  // end of test op ExpandDims
 
+// Test op: OneHot
+TEST(ArrayOps, OneHot1dNegAxis) {
+  Scope root = Scope::NewRootScope();
+
+  vector<int> static_input_indexes = {1};
+
+  Tensor indices(DT_INT32, TensorShape({4}));
+
+  AssignInputValues<int>(indices, {0, 2, -1, 1});
+
+  Tensor depth(DT_INT32, TensorShape({}));
+  Tensor on_value(DT_FLOAT, TensorShape({}));
+  Tensor off_value(DT_FLOAT, TensorShape({}));
+
+  AssignInputValues<int>(depth, 3);
+  AssignInputValues<float>(on_value, 5.0);
+  AssignInputValues<float>(off_value, 0.0);
+
+  auto attrs = ops::OneHot::Attrs();
+  attrs.axis_ = -1;
+
+  auto R = ops::OneHot(root, indices, depth, on_value, off_value, attrs);
+  vector<DataType> output_datatypes = {DT_FLOAT};
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "OneHot", static_input_indexes, output_datatypes,
+                        sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+TEST(ArrayOps, OneHot1d) {
+  Scope root = Scope::NewRootScope();
+
+  vector<int> static_input_indexes = {1};
+
+  Tensor indices(DT_INT32, TensorShape({4}));
+
+  AssignInputValues<int>(indices, {0, 2, -1, 1});
+
+  Tensor depth(DT_INT32, TensorShape({}));
+  Tensor on_value(DT_FLOAT, TensorShape({}));
+  Tensor off_value(DT_FLOAT, TensorShape({}));
+
+  AssignInputValues<int>(depth, 3);
+  AssignInputValues<float>(on_value, 5.0);
+  AssignInputValues<float>(off_value, 0.0);
+
+  auto attrs = ops::OneHot::Attrs();
+  attrs.axis_ = 0;
+
+  auto R = ops::OneHot(root, indices, depth, on_value, off_value, attrs);
+  vector<DataType> output_datatypes = {DT_FLOAT};
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "OneHot", static_input_indexes, output_datatypes,
+                        sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+TEST(ArrayOps, OneHot2dNegAxis) {
+  Scope root = Scope::NewRootScope();
+
+  vector<int> static_input_indexes = {1};
+
+  Tensor indices(DT_INT32, TensorShape({2, 2}));
+
+  AssignInputValues<int>(indices, {0, 2, -1, 1});
+
+  Tensor depth(DT_INT32, TensorShape({}));
+  Tensor on_value(DT_FLOAT, TensorShape({}));
+  Tensor off_value(DT_FLOAT, TensorShape({}));
+
+  AssignInputValues<int>(depth, 3);
+  AssignInputValues<float>(on_value, 5.0);
+  AssignInputValues<float>(off_value, 0.0);
+
+  auto attrs = ops::OneHot::Attrs();
+  attrs.axis_ = -1;
```

Comment:
I think we checked it, but can you please confirm that axis can only take values [-1, rank)

Reply:
i already verified this while writing the unit tests and right, axis can only take values -1 to rank.